### PR TITLE
Exported Killav into a post-exploitation module

### PR DESCRIPTION
**Pull request by madmantm**
I was unsure if this was the place to send the update, i will update as i find new antivirus or exe's that need killing.

### CODE REVIEW

**Comment by void-in** [comment]
Shouldn't the list of AVs be saved in a text file in data/ directory and then read from the module? I know the previous scripts you merged were having this list included in the module source as well. Just a thought so that future updates will require adding the executable names to the text file rather than changing the module. 

/cc @wchen-r7

**Comment by madmantm** [comment]
i will do the modification and post back.

**Comment by wchen-r7** [comment]
> Shouldn't the list of AVs be saved in a text file in data/ directory and then read from the module?

Yeah this would be a good idea. Easier to maintain, too. Good call. The text file probably would be in data/wordlists.

File: modules/post/windows/manage/killav.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        'License'       => MSF_LICENSE,
+        'Author'        => [ 'Marc-Andre Meloche (MadmanTM)', 'Nikhil Mittal (Samratashok)', 'Jerome Athias'],
+        'Platform'      => [ 'win' ],
+        'SessionTypes'  => [ 'meterpreter' ]
+    ))
+  end
+
+  def run
+
+avs = ::File.read(::File.join(Msf::Config.data_directory, "/wordlists/av_list.txt"))
```
Comment (OJ) [inline modules/post/windows/manage/killav.rb]: small indenting issue here. Also, you should let `File.join` to more heavy lifting:
``` ruby
avs = ::File.read(::File.join(Msf::Config.data_directory, 'wordlists', 'av_list.txt'))
```

File: modules/post/windows/manage/killav.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+   client.sys.process.get_processes().each do |x|
+     if (avs.index(x['name'].downcase))
```
Comment (OJ) [inline modules/post/windows/manage/killav.rb]: Extra parentheses not needed, also, would `include?` be better?:
``` ruby
if avs.include?(x['name'].downcase)
  ...
```

File: modules/post/windows/manage/killav.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+       print_status("Killing off #{x['name']}...")
+       client.sys.process.kill(x['pid'])
```
Comment (OJ) [inline modules/post/windows/manage/killav.rb]: If `client.sys.process.kill` fails, will this terminate the loop? If so, is this a problem (ie. what if they're running multiple AVs :8ball: )

Comment (madmantm) [inline modules/post/windows/manage/killav.rb]: well, with my tests, it kills all processes it detects, i tried it with mcafee and emet, and it will stop.

**Comment by OJ** [comment]
There were other issues with this implementation:

1. The file read resulted in a single string, rather than a list of strings. Use of `include?` or `index` would so a substring match, resulting in files matching that shouldn't match.
1. There were more indenting/style problems.
1. The module description was not really indicative of what the module attempted to do.
1. The lack of exception handling on the `kill` call would result in the module failing early, potentially missing out on other processes.
1. If no processes were found, the user wasn't informed of what attempted.

@madmantm The PR that I've just submitted to you at https://github.com/madmantm/metasploit-framework/pull/1 has a few changes to fix this stuff. Could you please test this on your set up and make sure it still functions? Thanks.

Sample output (of failure):
```
msf post(killav) > run

[*] Attempting to terminate 'ccSvcHst.exe' (PID: 1688) ...
[-] Failed to terminate 'ccSvcHst.exe' (PID: 1688).
[*] Attempting to terminate 'ccSvcHst.exe' (PID: 7240) ...
[-] Failed to terminate 'ccSvcHst.exe' (PID: 7240).
[+] A total of 2 process(es) were discovered, 0 were terminated.
[*] Post module execution completed
```

**Comment by OJ** [comment]
@void-in and @wchen-r7 would you guys mind taking a quick look please?

**Comment by wchen-r7** [comment]
@OJ that looks good. Exactly what I was thinking too. I'd put you as an author :-)

**Comment by madmantm** [comment]
I was about to propose that OJ was added to the list of authors! :)

**Comment by wchen-r7** [comment]
OK I'll try to land this today :-) Thanks all!

**Comment by madmantm** [comment]
Nice! :) Thanks!

**Comment by OJ** [comment]
Thanks Wei :)

**Pull request merged**